atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/CacheMetrics.java

```diff
@@ -0,0 +1,96 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api.cache;
+
+import com.palantir.atlasdb.AtlasDbMetricNames;
+import com.palantir.atlasdb.util.AccumulatingValueMetric;
+import com.palantir.atlasdb.util.CurrentValueMetric;
+import com.palantir.atlasdb.util.MetricsManager;
+
+public final class CacheMetrics {
+    private final AccumulatingValueMetric hits;
+    private final AccumulatingValueMetric misses;
+    private final AccumulatingValueMetric cacheSize;
+    private final CurrentValueMetric<Integer> eventCacheValidationFailures;
+    private final CurrentValueMetric<Integer> valueCacheValidationFailures;
+    private final MetricsManager metricsManager;
+
+    private CacheMetrics(
+            AccumulatingValueMetric hits,
+            AccumulatingValueMetric misses,
+            AccumulatingValueMetric cacheSize,
+            CurrentValueMetric<Integer> eventCacheValidationFailures,
+            CurrentValueMetric<Integer> valueCacheValidationFailures,
+            MetricsManager metricsManager) {
+        this.hits = hits;
+        this.misses = misses;
+        this.cacheSize = cacheSize;
+        this.eventCacheValidationFailures = eventCacheValidationFailures;
+        this.valueCacheValidationFailures = valueCacheValidationFailures;
+        this.metricsManager = metricsManager;
+    }
+
+    public static CacheMetrics create(MetricsManager metricsManager) {
+        return new CacheMetrics(
+                metricsManager.registerOrGetGauge(
+                        CacheMetrics.class, AtlasDbMetricNames.CACHE_HITS, AccumulatingValueMetric::new),
+                metricsManager.registerOrGetGauge(
+                        CacheMetrics.class, AtlasDbMetricNames.CACHE_MISSES, AccumulatingValueMetric::new),
+                metricsManager.registerOrGetGauge(
+                        CacheMetrics.class, AtlasDbMetricNames.CACHE_SIZE, AccumulatingValueMetric::new),
+                metricsManager.registerOrGetGauge(
+                        CacheMetrics.class,
+                        AtlasDbMetricNames.EVENT_CACHE_VALIDATION_FAILURES,
+                        CurrentValueMetric::new),
+                metricsManager.registerOrGetGauge(
+                        CacheMetrics.class,
+                        AtlasDbMetricNames.VALUE_CACHE_VALIDATION_FAILURES,
+                        CurrentValueMetric::new),
+                metricsManager);
+    }
+
+    public void registerHits(long number) {
+        hits.accumulateValue(number);
+    }
+
+    public void registerMisses(long number) {
+        misses.accumulateValue(number);
+    }
```

Comment:
The `AccumulatingValueMetric` is backed by an `AtomicLong`. How much cross-thread interaction do we expect? At some point it could make sense for us to use `LongAdder` if this is high

Reply:
Replaced with `Counter`